sa/sa.go

```diff
@@ -2143,3 +2149,81 @@ func (ssa *SQLStorageAuthority) IncidentsForSerial(ctx context.Context, req *sap
 	}
 	return incidentsForSerial, nil
 }
+
+// SerialsForIncident queries the provided incident table and returns the
+// resulting rows as a stream of `*sapb.IncidentSerial`s. An `io.EOF` error
+// signals that there are no more serials to send. If the incident table in
+// question contains zero rows, only an `io.EOF` error is returned.
+func (ssa *SQLStorageAuthority) SerialsForIncident(req *sapb.SerialsForIncidentRequest, stream sapb.StorageAuthority_SerialsForIncidentServer) error {
+	if req.IncidentTable == "" {
+		return errIncompleteRequest
+	}
+
+	// Check that `req.IncidentTable` is a valid incident table name.
+	if !validIncidentTableRegexp.MatchString(req.IncidentTable) {
+		return berrors.MalformedError("requested table name is malformed")
```

Comment:
MalformedError gets turned directly into a ProblemDetails, which implied that its contents are a user-visible error. If this method somehow gets an invalid table name, that's an "us" problem, not a "user" problem. For instance imagine this path is getting called from an ARI endpoint and we have a bug that's generating invalid table names. That should be an InternalServerError. We might also want to see some internal details in our logs, while giving a boring message to the user.

The way to accomplish that is to not use berrors at all. Instead, just return `fmt.Errorf("malformed table name %q", req.IncidentTable)`.

The semantics of `berrors` are: I expect this error to be treated specially, for instance by being turned into a specific ProblemDetails that I want to show the user, or by being handled internally (like NotFound).

Plain errors get handled in the WFE by getting turned into a user-facing serverInternal problem, and also by logging the internal details for our consumption.